src/full_client.toit

```diff
@@ -1079,10 +1079,13 @@ class FullClient:
     store can use this information to avoid keeping the data in memory, or to clear data from
     the flash.
   */
-  publish topic/string payload/ByteArray --qos/int=1 --retain/bool=false --persistence-token/any=null -> none:
+  publish topic/string payload/io.Data --qos/int=1 --retain/bool=false --persistence-token/any=null -> none:
+    byte-array-payload := payload is ByteArray
```

Comment:
Wouldn't it be better to forward the `io.Data` further down, so we can write it like that over the TCP socket?

Reply:
added a comment.